test/contracts/front/challenges.js

```diff
@@ -379,6 +387,14 @@ describe('Challenges Contract', () => {
       cooldown = parseInt(await getCooldown(), 10);
     });
 
+    it('Emits proper event', async () => {
+      const events = await observeEventEmission(web3, () => resolveChallenge(challengeId, resolver), challengesEventEmitter, 'ChallengeResolved');
+      expect(events.length).to.equal(1);
```

Comment:
A very similar pattern is repeated some 3 times in this file. Maybe a helper function?